plugins/module_utils/network/sonic/config/vlans/vlans.py

```diff
@@ -153,20 +172,25 @@ def _state_overridden(self, want, have, diff):
         :returns: the commands necessary to migrate the current configuration
                   to the desired configuration
         """
-        ret_requests = list()
-        commands = list()
-        vlans_to_delete = get_diff(have, want, TEST_KEYS)
-        if vlans_to_delete:
-            delete_vlans_requests = self.get_delete_vlans_requests(vlans_to_delete)
-            ret_requests.extend(delete_vlans_requests)
-            commands.extend(update_states(vlans_to_delete, "deleted"))
+        commands = []
+        requests = []
+
+        r_diff = get_diff(have, want, TEST_KEYS)
+        if have and (diff or r_diff):
+            del_requests = self.get_delete_vlans_requests(have, True)
+            requests.extend(del_requests)
+            commands.extend(update_states(have, "deleted"))
+            have = []
 
-        if diff:
-            vlans_to_create_requests = self.get_create_vlans_requests(diff)
-            ret_requests.extend(vlans_to_create_requests)
-            commands.extend(update_states(diff, "merged"))
+        if not have and want:
+            ovr_commands = want
+            ovr_requests = self.get_create_vlans_requests(ovr_commands)
```

Comment:
Modifying a single existing VLAN, would result in deleting all the VLANs configured in the device and reconfiguring them. This can be handled similar to 'replaced' state, wherein the VLANs that are in 'r_diff' and not in 'want' can be deleted and others are modified/added as in 'replaced'.

Reply:
Addressed.